integration-tests/gradle/src/test/resources/custom-java-native-sourceset-module/build.gradle

```diff
@@ -0,0 +1,53 @@
+plugins {
+    id 'java'
+    id 'io.quarkus'
+}
+
+repositories {
+    if (System.properties.containsKey('maven.repo.local')) {
+        maven {
+            url System.properties.get('maven.repo.local')
+        }
+    } else {
+        mavenLocal()
+    }
+    mavenCentral()
+}
+
+dependencies {
+    implementation enforcedPlatform("${quarkusPlatformGroupId}:${quarkusPlatformArtifactId}:${quarkusPlatformVersion}")
+    implementation 'io.quarkus:quarkus-resteasy'
+
+    testImplementation 'io.quarkus:quarkus-junit5'
+    testImplementation 'io.rest-assured:rest-assured'
+}
+
+group 'org.acme'
+version '1.0.0-SNAPSHOT'
+
+compileJava {
+    options.encoding = 'UTF-8'
+    options.compilerArgs << '-parameters'
+}
+
+compileTestJava {
+    options.encoding = 'UTF-8'
+}
+
+sourceSets {
```

Comment:
I think it makes more sense to have a new sourceSet defined here instead?

Reply:
The `integrationTest` sourceSet is a new source set. I don't understand which source set we should create?

Reply:
You said:

> This branch allows to configure an extra source set that will be added to the native-test source set.

Wouldn't it make sense to declare this extra source set here?

Reply:
the `native-test` source set? as we create and configure the `testNative` task, we need to create it in the plugin.

Reply:
Ah, I see. Do you think it makes sense to nest in a managed `sourceSets` element inside the plugin (for clarity and future-proof changes?). Something like:

```gradle
quarkus {
    sourceSets {
		extraNativeTest = sourceSets.integrationTest
    }
}
```

Reply:
yes we can, that's a good idea!